### ISSUE TITLE
New lint cop to check mistypes in `+=` or `-=`

### ISSUE THREAD

**Issue by fatkodima** [rubocop-hq/rubocop#9223]
Inspired by https://rules.sonarsource.com/ruby/type/Bug/RSPEC-2757 and https://www.viva64.com/en/w/v588/

I think many people did mistakes like the following
```ruby
x =+ 2
```

**Comment by Drenmi** [comment]
I guess the case of `x =- 2` is more likely to have false positives, because it could be `x = -2`, but the space went in the wrong place. 🙂 Either way, I think this is a solid candidate for a `Lint` cop.

**Comment by bbatsov** [comment]
I also think such a cop is a good idea!
